### PR TITLE
[qt][windows] Make Mapbox GL build on Windows with clang-cl

### PR DESCRIPTION
**Pull request by tmpsantos**
This should create a qmapboxgl.dll that is compatible with Qt 5 built with MSVC.

### CODE REVIEW

**Comment by tmpsantos** [comment]
Need to get https://www.appveyor.com/ CI working before merging this.

**Comment by brunoabinader** [comment]
@tmpsantos Fixed iOS build in 5c235364253407da119b4f5868bfdf9eaee88be2 by asking the compiler to ignore the warning coming from the newer Boost header (gotta ❤️ Boost).

File: include/mbgl/util/tileset.hpp [hunk 1]
```diff
@@ -37,6 +37,11 @@ class Tileset {
         return std::tie(lhs.tiles, lhs.zoomRange, lhs.attribution, lhs.scheme, lhs.bounds)
             == std::tie(rhs.tiles, rhs.zoomRange, rhs.attribution, rhs.scheme, rhs.bounds);
     }
+
+    friend bool operator!=(const Tileset& lhs, const Tileset& rhs) {
+        return std::tie(lhs.tiles, lhs.zoomRange, lhs.attribution, lhs.scheme)
```
Comment (brunoabinader) [inline include/mbgl/util/tileset.hpp]: What about this:
```C++
friend bool operator!=(const Tileset& lhs, const Tileset& rhs) {
    return !(lhs == rhs);
}
```

Comment (tmpsantos) [reply include/mbgl/util/tileset.hpp]: :+1:

**Comment by tmpsantos** [comment]
> Could you please also add a Windows entry in our README.md?

:+1:

**Pull request merged**